client/src/components/autograder/GradeDefinitionInputField.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import './GradeDefinitionInputField.less'
+import {
+  GradeDefinition,
+  GradeDefinitionInput,
+  useUpdateGradeDefinitionValuesMutation
+} from '../../generated/graphql'
+import HelpTooltip from '../HelpTooltip'
+import { debounce } from 'ts-debounce'
+import { valueType } from 'antd/es/statistic/utils'
+
+const renderGradePoints = (
+  title: string,
+  value: number,
+  onChange: (value: number) => void,
+  additionalProps: InputNumberProps = {}
+) => {
+  const onChangeValid = (val: valueType | undefined) => {
+    if (val !== undefined) {
+      if (typeof val !== 'string') {
+        onChange(val)
+      }
+    }
+  }
+
+  return (
+    <InputNumber
+      title={title}
+      min={0}
+      onChange={onChangeValid}
+      inputMode={'numeric'}
+      value={value}
+      {...additionalProps}
+    />
+  )
+}
+
+const renderGradeErrors = (
+  title: string,
+  value: number,
+  max: number,
+  onChange: (value: number) => void,
+  additionalProps: InputNumberProps = {}
+) => {
+  const onChangeValid = (val: valueType | undefined) => {
+    if (val !== undefined) {
+      if (typeof val !== 'string') {
+        onChange(val)
+      }
+    }
+  }
+
+  return (
+    <InputNumber
+      title={title}
+      min={0}
+      max={max}
+      onChange={onChangeValid}
+      inputMode={'numeric'}
+      value={value}
+      {...additionalProps}
+    />
+  )
+}
+
+/**
+ * Inputform to Enter and Mutate a GradeDefinition
+ *
+ */
+const GradeDefinitionInputField: React.FC<{
+  gradeDefinition: GradeDefinition
+  fetchForUpdate: any
+  disable: boolean
+}> = props => {
+  const [updateMutation] = useUpdateGradeDefinitionValuesMutation({
+    onCompleted: () => {
+      props.fetchForUpdate()
+    }
+  })
+
+  const [globalField, setGlobalField] = useState<
+    keyof GradeDefinitionInput | null
+  >(null)
+
+  const [gradeDefinition, setGradeDefinition] = useState<GradeDefinition>(
+    props.gradeDefinition || {
+      pEvalMax: 0,
+      bOnMinor: 0,
+      bOnMajor: 0,
+      bOnCritical: 0
+    }
+  )
+
+  const [input, setGradeDefinitionInput] = useState<GradeDefinitionInput>()
+  // TODO updates two times, due to gradeDefinition Dependency here and in useEffekt for memorizeCallback. Works but its not flawless.
+  useEffect(() => {
+    setGradeDefinitionInput({
+      criticalError: gradeDefinition.criticalError,
+      majorError: gradeDefinition.majorError,
+      minorError: gradeDefinition.minorError,
+      id: gradeDefinition.id,
+      maxPoints: gradeDefinition.maxPoints
+    })
+  }, [gradeDefinition])
+
+  const memoizedCallback = useCallback(() => {
+    if (globalField !== null) {
+      if (input !== undefined)
+        debounce(updateMutation({ variables: { input } }).then, 500)
+    }
+  }, [globalField, updateMutation, input])
+
+  useEffect(() => {
+    memoizedCallback()
+  }, [gradeDefinition, memoizedCallback])
```

Comment:
Whenever the Gradedefinition gets updated the useEffect Hook gets called and to prevent an infinite updateloop due to the gradeDefinition being in the dependency (tslint forced me to do this) i found a solution with memorizing the callback.